Update getting started section to include mention of processed objects

allyhawkins opened the pull request:
Closes #98 

Here I made some tweaks to the getting started section to mention that we now have a processed object and that users can directly work with that rather than the filtered object. I modified some of the language at the beginning to describe all three files now and suggested that users work with the processed object. I also updated the section on reading in files to use the processed object rather than the filtered object. 

The other thing I added here was a small section on working with the processed objects and had a few examples on how to access some of the things that are unique to these objects. In this section, I included a statement that these objects are ready for downstream analysis including clustering, but didn't provide a ton of detail. I wasn't sure how much we wanted to get into it. I did provide some resources on clustering from OSCA though. I also think we probably want to add a link to the clustering exploration module in ScPCA once that is able to take output from the updated `scpca-nf`. 

I then made a separate section on working with the filtered objects that contains the information that was previously in this section, so I didn't remove it. 

The last thing I did was update the mention of `scpca-downstream-analyses` to note that it includes general workflows for working with single-cell data, and not just mention of the core workflow which is what was there previously.

jashapiro commented on docs/getting_started.md:
```diff
@@ -31,17 +34,57 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
-## Quality control
+## Working with the processed `SingleCellExperiment` objects
+
+The `processed` objects contain both the raw and normalized gene expression data, so many of the steps described in the section below can be skipped.
```
This sentence seems oddly placed to me. I think you are referring to the "working with filtered" section, but that isn't really obvious. This can probably be removed with the link to the processing description moved up?

jashapiro commented on docs/getting_started.md:
````diff
@@ -31,17 +34,57 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
+normalized_counts <- logcounts(processed_sce)
+```
+
+These objects also contain a list of highly variable genes identified for the given library.
+These can be accessed using the following command:
+
+```r
+highly_variable_genes <- metadata(processed_sce)$highly_variable_genes
+```
````
I might put this after the PCA, as the initial point of this list is to identify the genes that were used for PCA. If these are ranked (they are, correct?) then we should also mention that this list is ordered, as that would be useful information for use outside that context.

jashapiro commented on docs/getting_started.md:
```diff
@@ -31,17 +34,57 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
+
+If you prefer to work with the `filtered` objects and perform the quality control and normalization yourself, see below for details on working with these objects.
+
+### Quality control
```
I think all this is fine to keep, but I wonder how much of it we really need it at this point? i.e. should we just assume that people who are using the filtered data know what they are doing?

jashapiro reviewed (commented):
Thanks for making these addtions! I had a few comments with little changes and a bit of organization...

allyhawkins commented:
I went ahead and cleaned this up a bit and did some rearranging based on your comments. I think I addressed everything except for the comment about removing some of the information on working with the filtered objects. I am going back and forth on this... although I think it is not as important now that we have the processed objects, I do think it could be useful for some users and since we already wrote it I don't think it hurts to keep.

jashapiro commented on docs/getting_started.md:
```diff
@@ -31,18 +36,60 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
+
+The `SingleCellExperiment` objects stored in the `_processed.rds` files have already undergone quality control to remove low quality cells, normalization, and dimensionality reduction such as principal component analysis and UMAP.
+
+The following commands can be used to access the raw and normalized counts matrices:
```
The double plural seems odd to me.
```suggestion
The following commands can be used to access the raw and normalized count matrices:
```

jashapiro commented on docs/getting_started.md:
```diff
@@ -31,18 +36,60 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
-## Quality control
+## Working with the processed `SingleCellExperiment` objects
+
+The `SingleCellExperiment` objects stored in the `_processed.rds` files have already undergone quality control to remove low quality cells, normalization, and dimensionality reduction such as principal component analysis and UMAP.
```
I broke this into two sentences, as it seemed a bit off
```suggestion
The `SingleCellExperiment` objects stored in the `processed.rds` files have undergone additional quality control steps to remove low quality cells.
In addition,  normalized expression counts and dimensionality reduction (principal component analysis and UMAP) have been calculated.
```

jashapiro commented on docs/getting_started.md:
```diff
@@ -2,17 +2,22 @@
-There are two RDS files that will be available for each library, a `filtered.rds` and an `unfiltered.rds` file.
-The `filtered.rds` files will contain the gene expression data for only droplets that are likely to contain cells, removing any potential empty droplets.
-The `unfiltered.rds` file will contain the gene expression data for all droplets, regardless of the presence of a cell or not.
+There are three RDS files that are available for each library: an `_unfiltered.rds`, a `filtered.rds`, and a `processed.rds` file.
```
I removed the lone underscore (which I may have inserted). I couldn't decide if we shoudl ahve those, but we don't use them in most places, so better to just leave them all off.
```suggestion
There are three RDS files that are available for each library: an `unfiltered.rds`, a `filtered.rds`, and a `processed.rds` file.
```

jashapiro commented on docs/getting_started.md:
```diff
@@ -2,17 +2,22 @@
 See the {ref}`section on filtering cells<processing_information:filtering cells>` for more information on how we remove potential empty droplets.
-In most scenarios, we recommend starting with the `filtered.rds` file.
+
+The `normalized.rds` files are further filtered to remove any low quality cells and contains both the raw and normalized gene expression data for the identified cells.
```
```suggestion
The `processed.rds` files are further filtered to remove any low quality cells and contain both the raw and normalized gene expression data for the identified cells.
```

jashapiro commented on docs/getting_started.md:
```diff
@@ -31,18 +36,60 @@ if (!("SingleCellExperiment" %in% installed.packages())) {
+Principal components were calculated from a set of highly variable genes identified for a given library.
+The list of highly variable genes used for this calculation, in order from highest to lowest variation, is stored in the `metadata` of the `SingleCellExperiment` object.
+
+These can be accessed using the following command:
```
```suggestion
This list can be accessed using the following command:
```

jashapiro reviewed (approved):
This looks good, just a few minor wording changes.
